### PR TITLE
Move replication related code to subfolder under history

### PR DESCRIPTION
**Pull request by yux0**
**What changed?**
Move replication related code to a new package in history

**Why?**
Refactor

**How did you test it?**
Existing unit tests

**Potential risks**
In the worst case, the replication stops working

### CODE REVIEW

File: service/history/replicationDLQHandler_mock.go [hunk 1]
```diff
@@ -1,105 +0,0 @@
-// The MIT License (MIT)
```
Comment (yycptt) [inline service/history/replicationDLQHandler_mock.go:1]: remove this file?

File: service/history/replicatorQueueProcessor.go [hunk 2]
```diff
@@ -18,13 +18,17 @@
...
+//TODO move this file to replication subfolder
+
 package history
 
 import (
 	ctx "context"
 	"errors"
 	"time"
 
+	"github.com/uber/cadence/service/history/replication"
```
Comment (yycptt) [inline service/history/replicatorQueueProcessor.go:30]: Group with cadence imports

File: service/history/replicationDLQHandler_mock.go (hunk 1 above)
Comment (yux0) [reply service/history/replicationDLQHandler_mock.go:1]: The new file is dlq_handler_mock.go

File: service/history/replicatorQueueProcessor.go (hunk 2 above)
Comment (yux0) [reply service/history/replicatorQueueProcessor.go]: done

File: service/history/replication/task_processor.go [hunk 3]
```diff
@@ -51,7 +51,8 @@ const (
 	replicationTimeout               = 30 * time.Second
 	taskErrorRetryBackoffCoefficient = 1.2
 	dlqErrorRetryWait                = time.Second
-	emptyMessageID                   = -1
+	// EmptyMessageID is the default start message ID
+	EmptyMessageID = -1
```
[the remark this answers is not in the document]
Comment (yux0) [reply service/history/replication/task_processor.go]: done

File: service/history/replication/task_fetcher.go [hunk 4]
```diff
@@ -56,39 +56,42 @@ type (
...
 		common.Daemon
 
 		GetSourceCluster() string
 		GetRequestChan() chan<- *request
 	}
 
-	// ReplicationTaskFetchers is a group of fetchers, one per source DC.
-	ReplicationTaskFetchers interface {
+	// TaskFetchers is a group of fetchers, one per source DC.
+	TaskFetchers interface {
```
[the remark this answers is not in the document]
Comment (yux0) [reply service/history/replication/task_fetcher.go:57]: done

File: service/history/replication/task_fetcher.go [hunk 5]
```diff
@@ -56,39 +56,42 @@ type (
 		requestChan    chan *request
 		done           chan struct{}
 	}
-	// ReplicationTaskFetcher is responsible for fetching replication messages from remote DC.
-	ReplicationTaskFetcher interface {
+	// TaskFetcher is responsible for fetching replication messages from remote DC.
+	TaskFetcher interface {
```
[the remark this answers is not in the document]
Comment (yux0) [reply service/history/replication/task_fetcher.go:49]: Done

File: service/history/historyEngine.go [hunk 6]
```diff
@@ -28,6 +28,8 @@ import (
 	"fmt"
 	"time"
 
+	"github.com/uber/cadence/service/history/replication"
```
[the remark this answers is not in the document]
Comment (yux0) [reply service/history/historyEngine.go]: Done

File: service/history/handler.go [hunk 7]
```diff
@@ -27,6 +27,8 @@ import (
 	"sync/atomic"
 	"time"
 
+	"github.com/uber/cadence/service/history/replication"
```
[the remark this answers is not in the document]
Comment (yux0) [reply service/history/handler.go]: Done

**Comment by yycptt** [comment]
#3181

**Pull request merged**